src/coreclr/jit/importercalls.cpp

```diff
@@ -6947,6 +6942,18 @@ class SpillRetExprHelper
 void Compiler::addFatPointerCandidate(GenTreeCall* call)
 {
     JITDUMP("Marking call [%06u] as fat pointer candidate\n", dspTreeID(call));
+
+    GenTree* fptr = call->gtCallAddr;
+    assert(fptr != nullptr);
+    if (!fptr->OperIsLocalRead())
+    {
+        // Spill the call address to a local for fat pointer handling
+        const unsigned fptrLclNum = lvaGrabTemp(true DEBUGARG("fat pointer temp"));
+        impStoreToTemp(fptrLclNum, fptr, CHECK_SPILL_ALL);
+        fptr             = gtNewLclvNode(fptrLclNum, genActualType(fptr->TypeGet()));
+        call->gtCallAddr = fptr;
+    }
```

Comment:
This is incorrect, it will cause the same problem as #121711.
This spilling needs to happen in `impImportCall` before args are popped, or the fix needs to look similar to #75634.